File: dgs-webflux/src/main/java/com/example/demo/gql/dataloaders/AuthorsDataLoader.java
package com.example.demo.gql.dataloaders;

import com.example.demo.gql.types.Author;
import com.example.demo.service.AuthorService;
import com.netflix.graphql.dgs.DgsDataLoader;
import lombok.RequiredArgsConstructor;
import org.dataloader.BatchLoader;

import java.util.List;
import java.util.concurrent.CompletionStage;

@RequiredArgsConstructor
@DgsDataLoader(name = "authorsLoader")
public class AuthorsDataLoader implements BatchLoader<Long, Author> {
    final AuthorService authorService;

    @Override
    public CompletionStage<List<Author>> load(List<Long> keys) {
        return this.authorService.getAuthorByIdIn(keys).collectList().toFuture();
    }
}

File: dgs-kotlin-co/src/main/kotlin/com/example/demo/gql/dataloader/AuthorsDataLoader.kt
package com.example.demo.gql.dataloader

import com.example.demo.gql.types.Author
import com.example.demo.service.AuthorService
import com.netflix.graphql.dgs.DgsDataLoader
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.asCoroutineDispatcher
import kotlinx.coroutines.flow.toList
import kotlinx.coroutines.future.future
import org.dataloader.BatchLoader
import java.util.UUID
import java.util.concurrent.CompletionStage
import java.util.concurrent.Executors

@DgsDataLoader(name = "authorsLoader")
class AuthorsDataLoader(val authorService: AuthorService) : BatchLoader<UUID, Author> {
    val loaderScope = CoroutineScope(Executors.newCachedThreadPool().asCoroutineDispatcher())
    override fun load(keys: List<UUID>): CompletionStage<List<Author>> = loaderScope.future {
        authorService.getAuthorByIdIn(keys).toList()
    }
}